src/middleware/headers.js

```diff
@@ -62,7 +62,7 @@ module.exports = function (middleware) {
 		}
 
 		if (process.env.NODE_ENV === 'development') {
-			headers['X-Upstream-Hostname'] = os.hostname();
+			headers['X-Upstream-Hostname'] = os.hostname().replace(/[^0-9A-Za-z-]/g, '');
```

Comment:
Should underscores and periods not be allowed?

Reply:
Chars based on https://man7.org/linux/man-pages/man7/hostname.7.html. Should have included a period as well (will be fixed in a new commit). Uppercase are technically not allowed either but didn't seem to break things.